cmd/influx/task.go

```diff
@@ -3,342 +3,425 @@ package main
 import (
 	"context"
 	"fmt"
-	"io"
-	"os"
 	"time"
 
+	"github.com/influxdata/influxdb/v2/tenant"
+
 	"github.com/influxdata/influxdb/v2"
-	"github.com/influxdata/influxdb/v2/cmd/influx/internal"
 	"github.com/influxdata/influxdb/v2/http"
 	"github.com/spf13/cobra"
 )
 
+type taskSVCsFn func() (influxdb.TaskService, influxdb.OrganizationService, error)
+
+func newTaskSVCs() (influxdb.TaskService, influxdb.OrganizationService, error) {
+	httpClient, err := newHTTPClient()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	orgSvc := &tenant.OrgClientService{Client: httpClient}
+	return &http.TaskService{Client: httpClient}, orgSvc, nil
+}
+
 func cmdTask(f *globalFlags, opt genericCLIOpts) *cobra.Command {
-	runE := func(cmd *cobra.Command, args []string) error {
-		seeHelp(cmd, args)
-		return nil
+	builder := newCmdTaskBuilder(newTaskSVCs, f, opt)
+	return builder.cmd()
+}
+
+type cmdTaskBuilder struct {
+	opts        genericCLIOpts
+	globalFlags *globalFlags
+
+	svcFn taskSVCsFn
+
+	taskID         string
+	runID          string
+	taskPrintFlags taskPrintFlags
+	// todo: fields of these flags structs could be pulled out for a more streamlined builder struct
```

Comment:
Ya this is for another PR. I feel like I would double reference the flag structs in other commands, so I left it alone. But I did pull out like taskID, runID, and org since those were pretty repeated